modules/medianetAnalyticsAdapter.js

```diff
@@ -382,6 +384,64 @@ class Auction {
   }
 }
 
+/**
+ * Here the idea is
+ * find all network entries via performance.getEntriesByType()
+ * filter it by video cache key in the url
+ * and exclude the ad server urls so that we dont match twice
+ * eg:
+ * dfp ads call: https://securepubads.g.doubleclick.net/gampad/ads?...hb_cache_id%3D55e85cd3-6ea4-4469-b890-84241816b131%26...
+ * prebid cache url: https://prebid.adnxs.com/pbc/v1/cache?uuid=55e85cd3-6ea4-4469-b890-84241816b131
+ *
+ * if the entry exists, emit the BID_WON
+ */
+function findRenderedVideoImpressions(auction) {
+  // filter for video bids
+  const videoBids = auction.bids.filter(bid => bid.mediaType === VIDEO && bid.context !== OUTSTREAM && bid.videoCacheKey)
+  // check if performance api is available
+  if (!videoBids.length || !window.performance || !window.performance.getEntriesByType) {
+    return;
+  }
+  const timeLimit = 15 * 1000 * 60;
+  const start = Date.now();
+  const bidWonMap = {};
+
+  let lastRead = 0;
+
+  function poll() {
+    // get network entries using the last read offset
+    const entries = window.performance.getEntriesByType('resource').splice(lastRead);
+    for (const res of entries) {
+      const url = res.name;
+      videoBids.forEach((bid) => {
+        // match the video cache key excluding ad server call
+        const matches = !url.includes(CONSTANTS.TARGETING_KEYS.CACHE_ID) && url.includes(bid.videoCacheKey);
+        if (matches && !bidWonMap[bid.bidId]) {
+          // video found
+          bidWonMap[bid.bidId] = bid;
+          bid.requestId = bid.bidId;
+          bid.auctionId = auction.acid;
+          bid.adUnitCode = bid.supplyAdCode;
+          // log BID_WON
+          bidWonHandler(bid);
+        }
+      });
+    }
+    // update offset
+    lastRead += entries.length;
+
+    const lastCheckedDiff = Date.now() - start;
+    if (lastCheckedDiff < timeLimit && Object.keys(bidWonMap).length < auction.adSlots.length) {
+      setTimeout(poll, 2e3);
+    }
+  }
+
+  // start polling for network entries
+  setTimeout(poll, 2e3);
```

Comment:
Also, can we improve logic here, rather than checking after every 2 seconds can we check more frequently?
IMO, 2 sec is a long duration which may lead discrepancies as user moves to next page during  2 sec gap.